src/Models/BaseElement.php

```diff
@@ -1005,4 +1040,22 @@ public function EvenOdd()
 
         return  ($odd) ? 'odd' : 'even';
     }
+
+    /**
+     * @param bool $disable
+     * @return $this
+     */
+    public function setDisableInlineEditing($disable = true)
+    {
+        $this->disableInlineEditing = (bool) $disable;
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function getDisableInlineEditing()
+    {
+        return (bool) $this->disableInlineEditing;
```

Comment:
Is there a way we can avoid having this temporary state property on `BaseElement`. I can see bugs in the future where strange things could happen if the in-line edit setting isn't toggle back on. Can we add `$directLinkOnly` or something as a parameter to `CMSEditLink`?